<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Post2dWindowParticleUnstructuredSettingDialog</class>
 <widget class="QDialog" name="Post2dWindowParticleUnstructuredSettingDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>482</width>
    <height>536</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Particle Setting</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_6">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_12">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QGroupBox" name="commonSettingsGroupBox">
         <property name="title">
          <string>Common Settings</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="0,1">
            <item>
             <widget class="QLabel" name="physValLabel">
              <property name="text">
               <string>Physical Value:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="solutionComboBox"/>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QRadioButton" name="periodicalRadioButton">
            <property name="text">
             <string>Generate periodically</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox">
            <property name="title">
             <string>Generation interval</string>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_2">
               <item>
                <widget class="TooltipLabel" name="timeHelpLabel">
                 <property name="enabled">
                  <bool>true</bool>
                 </property>
                 <property name="toolTip">
                  <string extracomment="If you specify 1, particles are generated for every time steps. If you specify 2, particles are generated twice equally separated, for every time steps, If you specify 1/2, particles are generated for every other time steps.">&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap;}
p {margin-top: 3px; margin-bottom: 3px;}
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS UI Gothic'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p&gt;If you specify 1, particles are generated for every time steps.&lt;/p&gt;
&lt;p&gt;If you specify 2, particles are generated twice equally separated, for every time steps,&lt;/p&gt;
&lt;p&gt;If you specify 1/2, particles are generated for every other time steps.&lt;/p&gt;
&lt;/body&gt;&lt;/html&gt;</string>
                 </property>
                 <property name="text">
                  <string/>
                 </property>
                 <property name="pixmap">
                  <pixmap resource="../../../guibase/guibase.qrc">:/libs/guibase/images/iconQuestion.png</pixmap>
                 </property>
                </widget>
               </item>
               <item>
                <spacer name="verticalSpacer">
                 <property name="orientation">
                  <enum>Qt::Vertical</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>20</width>
                   <height>40</height>
                  </size>
                 </property>
                </spacer>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QGridLayout" name="gridLayout_2">
               <item row="0" column="0">
                <widget class="QLabel" name="timeLabel_minus">
                 <property name="text">
                  <string>-</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QSlider" name="timeSlider">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="tickPosition">
                  <enum>QSlider::TicksBelow</enum>
                 </property>
                </widget>
               </item>
               <item row="0" column="2">
                <widget class="QLabel" name="timeLabel_plus">
                 <property name="text">
                  <string>+</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="3">
                <widget class="QLabel" name="timeValueLabel">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="minimumSize">
                  <size>
                   <width>41</width>
                   <height>21</height>
                  </size>
                 </property>
                 <property name="frameShape">
                  <enum>QFrame::Panel</enum>
                 </property>
                 <property name="frameShadow">
                  <enum>QFrame::Sunken</enum>
                 </property>
                 <property name="text">
                  <string/>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                 <property name="margin">
                  <number>2</number>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <layout class="QHBoxLayout" name="horizontalLayout_7">
                 <item>
                  <widget class="QLabel" name="timeMinLabel">
                   <property name="font">
                    <font>
                     <pointsize>8</pointsize>
                    </font>
                   </property>
                   <property name="text">
                    <string>min</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <spacer name="horizontalSpacer_8">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>40</width>
                     <height>20</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                 <item>
                  <widget class="QLabel" name="timeMidLabel">
                   <property name="font">
                    <font>
                     <pointsize>8</pointsize>
                    </font>
                   </property>
                   <property name="text">
                    <string>1</string>
                   </property>
                   <property name="alignment">
                    <set>Qt::AlignCenter</set>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <spacer name="horizontalSpacer_9">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>40</width>
                     <height>20</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                 <item>
                  <widget class="QLabel" name="timeMaxLabel">
                   <property name="font">
                    <font>
                     <pointsize>8</pointsize>
                    </font>
                   </property>
                   <property name="text">
                    <string>max</string>
                   </property>
                   <property name="alignment">
                    <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_11">
            <item>
             <widget class="QRadioButton" name="arbitraryRadioButton">
              <property name="text">
               <string>Generate at arbitrary time</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="arbitraryEditButton">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Edit...</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_4">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>18</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_9">
         <item>
          <widget class="QListWidget" name="startPositionListWidget">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>40</width>
             <height>16777215</height>
            </size>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="startPointsGroupBox">
           <property name="title">
            <string>Start Points</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QLabel" name="startPointDescLabel">
              <property name="text">
               <string>Specify two points. Particles start from points generated equally separated between these two points.</string>
              </property>
              <property name="wordWrap">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="pointsMouseRadioButton">
              <property name="text">
               <string>Mouse Drag</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="pointsValuesRadioButton">
              <property name="text">
               <string>Coordinate Values</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <spacer name="horizontalSpacer_3">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeType">
                 <enum>QSizePolicy::Fixed</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>10</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <layout class="QGridLayout" name="gridLayout">
                <item row="0" column="0">
                 <widget class="QLabel" name="point1Label">
                  <property name="text">
                   <string>Point 1:</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="RealNumberEditWidget" name="point1XEdit">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                 </widget>
                </item>
                <item row="0" column="2">
                 <widget class="QLabel" name="point1CommaLabel">
                  <property name="text">
                   <string>,</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="3">
                 <widget class="RealNumberEditWidget" name="point1YEdit">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="point2Label">
                  <property name="text">
                   <string>Point 2:</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="RealNumberEditWidget" name="point2XEdit">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                 </widget>
                </item>
                <item row="1" column="2">
                 <widget class="QLabel" name="point2CommaLabel">
                  <property name="text">
                   <string>,</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="3">
                 <widget class="RealNumberEditWidget" name="point2YEdit">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <spacer name="horizontalSpacer_5">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <widget class="QLabel" name="numPointsLabel">
                <property name="text">
                 <string>Number of points:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="numPointsSpinBox">
                <property name="minimum">
                 <number>2</number>
                </property>
                <property name="maximum">
                 <number>1000</number>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_8" stretch="0,1">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_5">
                <item>
                 <widget class="QLabel" name="colorLabel">
                  <property name="text">
                   <string>Color:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="ColorEditWidget" name="colorWidget"/>
                </item>
                <item>
                 <spacer name="horizontalSpacer_6">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_6">
                <item>
                 <widget class="QLabel" name="sizeLabel">
                  <property name="text">
                   <string>Size:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="TooltipLabel" name="sizeHelpLabel">
                  <property name="toolTip">
                   <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS UI Gothic'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:3px; margin-bottom:3px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Specify a number of pixels on display.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                  </property>
                  <property name="text">
                   <string/>
                  </property>
                  <property name="pixmap">
                   <pixmap resource="../../../guibase/guibase.qrc">:/libs/guibase/images/iconQuestion.png</pixmap>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QSpinBox" name="sizeSpinBox">
                  <property name="minimum">
                   <number>1</number>
                  </property>
                  <property name="maximum">
                   <number>20</number>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="horizontalSpacer_11">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_10">
         <item>
          <widget class="QPushButton" name="addPushButton">
           <property name="text">
            <string>&amp;Add</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="removePushButton">
           <property name="text">
            <string>&amp;Remove</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_10">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>4</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QPushButton" name="regionSettingButton">
         <property name="font">
          <font>
           <pointsize>12</pointsize>
          </font>
         </property>
         <property name="text">
          <string>&amp;Region Setting</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Apply|QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TooltipLabel</class>
   <extends>QLabel</extends>
   <header>guibase/tooltiplabel.h</header>
  </customwidget>
  <customwidget>
   <class>RealNumberEditWidget</class>
   <extends>QLineEdit</extends>
   <header>guibase/widget/realnumbereditwidget.h</header>
  </customwidget>
  <customwidget>
   <class>ColorEditWidget</class>
   <extends>QLabel</extends>
   <header>guibase/widget/coloreditwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../guibase/guibase.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>Post2dWindowParticleUnstructuredSettingDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>174</x>
     <y>526</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>Post2dWindowParticleUnstructuredSettingDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>174</x>
     <y>526</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pointsValuesRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>point1XEdit</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>342</x>
     <y>328</y>
    </hint>
    <hint type="destinationlabel">
     <x>224</x>
     <y>356</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pointsValuesRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>point1YEdit</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>342</x>
     <y>328</y>
    </hint>
    <hint type="destinationlabel">
     <x>336</x>
     <y>356</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pointsValuesRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>point2XEdit</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>342</x>
     <y>328</y>
    </hint>
    <hint type="destinationlabel">
     <x>224</x>
     <y>382</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pointsValuesRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>point2YEdit</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>342</x>
     <y>328</y>
    </hint>
    <hint type="destinationlabel">
     <x>336</x>
     <y>382</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>periodicalRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>groupBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>65</x>
     <y>68</y>
    </hint>
    <hint type="destinationlabel">
     <x>65</x>
     <y>84</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>arbitraryRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>arbitraryEditButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>147</x>
     <y>197</y>
    </hint>
    <hint type="destinationlabel">
     <x>213</x>
     <y>196</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
